doc/main/reference/try_put_and_wait.rst

```diff
@@ -0,0 +1,324 @@
+.. _try_put_and_wait:
+
+Waiting for single messages in Flow Graph
+=========================================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` to the receiving nodes in the Flow Graph.
+This function allows to wait until completion of single message, submitted to ``try_put_and_wait`` as an input by the Flow Graph in
+opposite to ``graph.wait_for_all()`` function that waits for completion of all tasks in the graph.
+
+The ``node.try_put_and_wait(msg)`` is performing ``node.try_put(msg)`` on the node and blocks until the work on ``msg`` is completed.
+It means the following two statements:
+
+* Any task spawned by any node in the Flow Graph that corresponds on working with ``msg`` or any other intermediate result
```

Comment:
```suggestion
* Any task initiated by any node in the Flow Graph that involves working with msg or any intermediate result derived from ``msg`` is completed.
```

Reply:
Partially fixed, using `derived from` can be ambiguous with the C++ inheritance terminology. Can we keep `computed` here?

Reply:
yes, computed is fine